### PR TITLE
490 update final report

### LINKED ISSUE #490: Update DFDs and improve final report
- Update DFDs.
- Update the Functional Requirements/User Stories, Description of MentorMatch, and User Groups (currently the final document has descriptions from the first milestone, and since then the client has adjusted the scope of certain aspects of the project so these sections should be updated).

### PR DESCRIPTION
**Pull request by katmartel**
Please note: It might be easier to read the pdf-report.md file by going to the file itself, rather than reading it from the 'files changed tab' (or you can mainly focus on the lines added so you can make comments easier. The flow of the document was sort of butchered by viewing the file in the 'files changed' tab, so I wanted to provide a little disclaimer here). 

On the DFD level 1 diagram, I removed the coloured rectangles which indicated what we planned to accomplish by the peer testing sessions since those project phases have passed. Please let me know if you disagree with this decision, and I'll make modifications as suggested. 

I wrote the 'example scenario for each user group of how they will use the software' with the intention of possibly using this for the 'promo video' (i.e Members of our team could be assigned to being either a Mentee, Mentor, or Admin user. We could read the little example blurbs and screen record the system pertaining to what that user can do.). I'll continue to give this some thought since there might be a more creative and/or interesting way to go about the promo video, I'm definitely open to ideas, this is just my initial thought.

### CODE REVIEW

**Review by aleksssk — APPROVED** [review]
Nice work!!!

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
The software that we have created over the duration of this project is a web-based Mentorship platform. The goal of this system is to facilitate the matching process between users seeking to learn skills (Mentees) and individuals who can offer guidance in sought after skills (Mentors). At the heart of this project are the Organizations, represented by one or more Admins, who host instances of the Mentee form on their public facing website and request Mentors internally from their part of our website: the admin's account. Suitable matches are created based on the number of similar skillsets and time-zone. proximity. Mentors can only be matched with Mentees who signed up on the website of the organization a mentor is part of. These matches are displayed as requests to Mentors, who then have the option to accept or deny requests. Once a Mentor accepts a request, the accepted Mentee is notified, allowing the two users to communicate, and for the Mentoring to begin.
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
Admins register their Organization by creating an account on our platform. To register, they need to provide the name of the Organization that they're from, as well as their name, email address, and a password. Once registered, an Admin can invite Mentors to join MentorMatch, and see which Mentor requests are pending as well as which Mentors are actively using the system. Admins can view the Mentor Form that requested Mentors receive the link to in an email upon being requested, as well as the Mentee Form, which they can generate an iFrame tag for to embed on their Organization's public website.
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
After learning about MentorMatch through a friend, and talking to individuals within our Organization, I decided to register. MentorMatch facilitates the process of bringing together skilled individuals from our organization so that they can aide in our vision of teaching skills to others. By going to MentorMatch and clicking on the 'create account' button, I'm able to register my organization and become the admin account holder. Once I have verified my email address by clicking a link I receive, I can manage my Organization's mentors by viewing who is a Mentor on the platform, as well as which of the Mentors that I have requested haven't yet registered. Requesting Mentors is simple, as only providing a valid email address is required to notify prospective Mentors that I'd like them to register on the platform.
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
Once requested by an Admin, Mentor users will receive an email with a link which brings them to that Organization's Mentor Form. Through this link, the user becomes associated with that Organization upon successful form submission. On that form, Mentors provide information about the skills they are able to teach, when they're available to mentor via a calendar link, and some basic information about themselves such as their name, email address, and a description. To be able to create their account, a unique email address must be provided. If the form has been filled out according to specifications, the user will be required to verify their email address before they can access Mentor functionalities within MentorMatch. Verified Mentors can view mentee requests and decide whether to accept or decline them. Though MentorMatch initially puts Mentors and Mentees in contact with each other, they then communicate outside of MentorMatch.
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
- As an individual currently working for an Organization registered on MentorMatch, I have experience in valuable skillsets that I would like to share with others. After receiving an email invite to join MentorMatch by my Organization's Admin, I am taken to a form which contains intuitive labels to indicate what information is required of me and offers tooltips and useful warnings in case I get stuck. Having the ability to provide a calendar link to my preferred Google calendar allows for flexibility in my choice of hours of availability, since these changes can be reflected on my calendar by booking times for confirmed Mentees, ensuring double-booking does not happen. After filling out the form and verifying my email, I can read a tutorial which gives a brief introduction to what I'm able to do as a Mentor. The "Mentee Requests" feature offers flexibility, as it allows me to decide whether I'd like to accept or decline a request based on the skills the Mentee is seeking to learn. The 'Accept Mentees' tab provides me with a reminder of the Mentors whose request I have already accepted, as well as basic information about each Mentee so that I can tailor my mentorship sessions towards their needs.
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
Mentees register through the Mentee Form, which can be accessed through an Organization's public website. Our system does not require mentees to have their own account. The Mentee Form prompts users to provide the skills that they want to be Mentored in as well as some basic information about themselves. Once the form has been submitted, a welcome email will be sent to the email address provided, informing the user that they will be notified once a Mentor has accepted their request. Based on the information provided on the form, MentorMatch will suggest Mentees to Mentors, considering common skills and time zones. After a Mentor accepts a Mentee request, that Mentee will receive an email containing information about the Mentor who accepted them, as well as the link which displays the Mentor's availability. From there, both users can correspond through email to arrange meetings.
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
- After browsing through an Organization's website, I noticed that the Organization is registered on MentorMatch. Since I can relate to the Organization's mission, and I want to further my knowledge in the skillsets that they offer, I decide to register. The Mentee registration form is embedded within the organization's website, and by following the tooltips and fixing any errors that are displayed, I submit the form, which notifies me that my submission has been successful, and that I'll be routed to MentorMatch in a new tab. Navigating to the new tab allows me to explore MentorMatch to see additional information. After checking my emails, I see one from MentorMatch welcoming me to the platform and letting me know that I'll be notified once a Mentor accepts my request.
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]: "Navigating to their calendar allows me to book an available time to meet." I don't think that's possible, [public calendars are read-only unless you explicitly add someone to give write access to](https://support.google.com/calendar/answer/37083?hl=en).

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
FR3. Once the user submits the publicly hosted Mentee Form, they will be notified whether their submission was successful. A new tab will then be opened to MentorMatch so that the user can chose whether they want to remain on their Organization's website or explore MentorMatch's website.
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
FR6. The MentorMatch login page will host 3 functionalities: Login, Forgot Password, and Create Account. A member of an organization will click on the Create Account button to access the Organization Form to register their Organization and create an account within the system.
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
FR7. To register Organizations and Mentors successfully, a unique email must be entered. If that email already exists in the database, a warning message will be displayed to the user and they will be blocked from signing up with that email.
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
FR8. Unique email addresses will be used as the user's username. They will also need to provide a password (containing at least 8 characters).
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
FR9. Logged in Admins will be able to publicly host the Mentee Form by clicking on the 'Generate iFrame tag' button on the 'View Mentee Form' tab, copying the piece of code that appears, and pasting it into their organization's website's source code.
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
FR12. Logged in Admins will be able to view the Mentor Form.
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
FR13. Once an Admin requests a Mentor using the mentor's email address, the Mentor will receive an email with a link to the Admin's Organization's Mentor Form.
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
FR14. Create a static Mentor Form with name, email, time zone, calendar link, password, confirm password, top three skills, and description fields.
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
FR19. Mentors and Admins will be able to reset their password by clicking on the “Forgot your password?” link on the Login page. The user will be required to enter an email address which has been associated with an account in the database. If successful, an email will be sent to that address, with a link which the user can click to be brought to a separate page to reset their password.
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
FR20. Mentors and Admins will be able to logout of their account, taking them to the login page, where they can login, retrieve their forgotten password, or register.
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]:
```suggestion
FR22. Once a Mentor has accepted a Mentee, an email will be sent to the Mentee to notify them of a successful match. The two can then correspond through email, and the Mentee will be able to view the mentor's availability via the Mentor's calendar link.
```

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [inline documentation/final-deliverables/pdf-report.md]: Again, I am afraid this is impossible, unless I am misunderstanding the Google Calendar documentation, or we assume that mentors add each mentee to their calendar manually.

**Review by jpabadir — APPROVED** [review]
Thanks for the good documentation! I added some suggestions on a few sentences where I thought we could improve the explanation, feel free to reply to any of them on the PR if you disagree, I'll make sure to read your replies.

File: documentation/final-deliverables/pdf-report.md
Comment (KatrinaMart) [reply documentation/final-deliverables/pdf-report.md:98]: My bad, poor wording on my end. I meant that by going to the Google Calendar, an individual can click on an event on the calendar, and provide a message stating that they'd like to be mentored at that time (I've attached a screenshot of this below to help visualize what I'm referring to). I can definitely see how this wasn't reflected in how I originally phrased things! Mentors having to manually update their calendars to reflect booked time
slots is definitely a limitation, but [this](https://support.google.com/calendar/answer/190998?co=GENIE.Platform%3DDesktop&hl=en) provides a simple fix that we can note in our documentation later.

Screen Shot 2021-03-23 at 3 39 11 PM
Screen Shot 2021-03-23 at 3 40 16 PM


For simplicity, would the following modification be alright?
```suggestion
  By checking my emails, I'm eventually notified that a Mentor has accepted my request. Within this email, I can see the Mentor's name, their email address, a description about the individual, and a link to their calendar. Navigating to their calendar allows me to view the Mentor's availability so that I can request a time to be mentored.
```

File: documentation/final-deliverables/pdf-report.md
Comment (katmartel) [reply documentation/final-deliverables/pdf-report.md]: My bad, poor wording on my end. I meant that by going to the Google Calendar, an individual can click on an event on the calendar, and provide a message stating that they'd like to be mentored at that time (I've attached a screenshot of this below to help visualize what I'm referring to). I can definitely see how this wasn't reflected in how I originally phrased things! Mentors having to manually update their calendars to reflect booked time
slots is definitely a limitation, but [this](https://support.google.com/calendar/answer/190998?co=GENIE.Platform%3DDesktop&hl=en) provides a simple fix that we can note in our documentation later.

Screen Shot 2021-03-23 at 3 39 11 PM
Screen Shot 2021-03-23 at 3 40 16 PM


For simplicity, would the following modification be alright?
```suggestion
  By checking my emails, I'm eventually notified that a Mentor has accepted my request. Within this email, I can see the Mentor's name, their email address, a description about the individual, and a link to their calendar. Navigating to their calendar allows me to view the Mentor's availability so that I can request a time to be mentored.
```

File: documentation/final-deliverables/pdf-report.md
Comment (katmartel) [reply documentation/final-deliverables/pdf-report.md]: I debated whether or not I should keep this FR tbh. The client hasn't mentioned anything about how Mentors and Mentees should unmatch within our system since the creation of this FR back in October when we made the 'Requirements Report' milestone (and its inclusion was more an aside that was mentioned when obtaining the scope of the project). Given this, I could possibly remove this FR for now, and ask the client about it during tomorrow's meeting?

**Comment by katmartel** [comment]
> Thanks for the good documentation! I added some suggestions on a few sentences where I thought we could improve the explanation, feel free to reply to any of them on the PR if you disagree, I'll make sure to read your replies.

Thanks for the feedback! :) I've incorporated the changes that you've suggested, and left comments on two of your suggestions to further elaborate. I'll merge this PR once you've had the chance to take a look at this so those two comments can be resolved.

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [reply documentation/final-deliverables/pdf-report.md]: How about change it to 
```suggestion
FR23. To "unmatch" Mentees and Mentors, the Mentee can communicate with the Mentor in order to cancel all future appointments, which the Mentor can reflect on his / her calendar. 
```

I think either that or remove it, since our system just doesn't support this.

File: documentation/final-deliverables/pdf-report.md
Comment (jpabadir) [reply documentation/final-deliverables/pdf-report.md]: Perfect!

**Review by jpabadir — APPROVED** [review]

File: documentation/final-deliverables/pdf-report.md
Comment (katmartel) [reply documentation/final-deliverables/pdf-report.md]: I'll leave it for now (with the changes that you've suggested). I'll mention it to the client tomorrow during the meeting just to double check that 'unmatching' isn't an important feature (which again, it hasn't been mentioned since October, so I doubt it's important).

**Pull request merged**
